gammapy/maps/wcs/geom.py

```diff
@@ -585,6 +585,22 @@ def _get_pix_all(
     def get_pix(self, idx=None, mode="center"):
         """Get map pixel coordinates from the geometry.
 
+        Parameters
+        ----------
+        mode : {'center', 'edges'}, optional
+            Get center or edge pix coordinates for the spatial axes.
+            Default is "center".
+
+        Returns
+        -------
+        coord : tuple
+            Map pixel coordinate tuple.
+        """
+        return self._pix(idx=idx, mode=mode)
```

Comment:
```suggestion
        return self._get_pix(idx=idx, mode=mode)
```

Reply:
this should break so this function is not use all all
could be simply `pass`

Reply:
Moreover I don't see any improvement, a reference to self is still give to lru_cache so the leak is still there